runtime/src/main/java/org/corfudb/runtime/LogReplicationLogicalGroupClient.java

```diff
@@ -0,0 +1,391 @@
+                } catch (TransactionAbortedException tae) {
+                    log.error("[{}] {}: {}", clientName, "Unable to remove destination, will be retried", tae);
+                    throw new RetryNeededException();
+                }
+            }).run();
+        } catch (InterruptedException e) {
+            log.error("[{}] {}: {}", clientName, "Unable to remove destination", e);
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Remove a list of destinations from the LogReplicationModelMetadataTable for a specific group.
+     *
+     * @param logicalGroup The group of tables that are replicated, represented as a string.
+     * @param remoteDestinations The cluster from which the logicalGroup should be removed from all future replication.
+     */
+    public void removeDestination(String logicalGroup, List<String> remoteDestinations) {
+        Preconditions.checkArgument(isValid(logicalGroup), String.format("[%s] %s", clientName, "logicalGroup is null or empty."));
+        Preconditions.checkArgument(isValid(remoteDestinations), String.format("[%s] %s", clientName, "remoteDestinations is null or empty."));
+        Preconditions.checkArgument(hasNoNullOrEmptyElements(remoteDestinations), String.format("[%s] %s", clientName, "remoteDestinations contains null or empty elements."));
+        Set<String> removeRemoteDestinationsSet = deduplicate(remoteDestinations);
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientDestinationInfoKey clientInfoKey = buildClientDestinationInfoKey(logicalGroup);
+                    DestinationInfoVal clientDestinations = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+
+                    if (clientDestinations != null) {
+                        Set<String> clientDestinationIdsSet = new HashSet<>(clientDestinations.getDestinationIdsList());
+                        List<String> clientDestinationIdsList = clientDestinationIdsSet.stream()
+                                .filter(i -> !removeRemoteDestinationsSet.contains(i))
+                                .collect(Collectors.toList());
+
+                        if (clientDestinationIdsList.size() == clientDestinationIdsSet.size()) {
+                            log.debug("[{}] {} {}", clientName, "No matching destinations found for:", removeRemoteDestinationsSet);
+                            return null;
+                        } else if (clientDestinationIdsList.size() + removeRemoteDestinationsSet.size() > clientDestinationIdsSet.size()) {
+                            removeRemoteDestinationsSet.removeAll(clientDestinationIdsSet);
+                            log.debug("[{}] {} {}", clientName, "Following destinations to remove are not present:", removeRemoteDestinationsSet);
```

Comment:
this case does not mean that all destinations in removeRemoteDestinationsSet are not present, right?

Reply:
correct, that is handled when we check that the sizes are equal, and now clientDestinationIdsList.size() < clientDestinationIdsSet.size() with latest commit.